chapters/en/slides/module4/module4_01.Rmd

```diff
@@ -0,0 +1,345 @@
+to two-dimensional and categorical data.
+
+---
+
+## Reading in the data
+
+```{python}
+import altair as alt
+from vega_datasets import data
+
+
+movies_extended = (
+    data.movies()
+    .dropna(subset=['Major Genre', 'MPAA Rating', 'Running Time min'])
+    .query('`MPAA Rating` != ["Not Rated", "NC-17"]'))
+movies_extended
+```
+
+We're continuing to work with the movies data set,
+but in contrast to last module we're filtering out some categories
+that contains few or problematic values.
+TODO filter the dataset beforehand?
+
+---
+
+## Visualizing a single numerical columns with points leads to overplotting
+
+```{python}
+alt.Chart(movies_extended).mark_point().encode(
+    alt.X('Production Budget'))
+```
+
+Notes:
+In the last slide deck we saw how to visualize the distribution
+of a single numerical dataframe column.
+What if we instead want to compare the distributions
+of two columns with each other?
+
+A question we could answer with this type of comparison is
+"Do higer grossing movies tend to have a high production budget?"
+
+Let's start the same way we did when visualizing a single distribution,
+by laying out all the individual observations as points across
+an axis.
+
+We can see that the there are so many observation that this plot is saturated,
+and we can't tell if there are more movies with a production budget of 80 million or 0.
+
+As we learned previously,
+a histogram, density plot, or boxplot would help us here
+if we were only interested in visualizing this single numerical column.
+
+---
+
+## Visualizing two numerical columns with points leads to overplotting
+
+```{python}
+alt.Chart(movies_extended).mark_point().encode(
+    alt.X('Production Budget'),
+    alt.Y('Worldwide Gross'))
+```
+
+Notes:
+Since we are interested in comparing two columns,
+we will add the second column to the y-axis.
+
+Sometimes this is enough to avoid overplotting
+since we are now spreading the points across two dimensions
+instead of one.
+
+However,
+as you can see in the plot on this slide,
+it is still impossible to tell
+if there are more points close to 80 million of 0,
+and likewise 200 million or 0 on the y-axis.
+
+So although we can discern a trend
+for the points outside the saturated area
+we do not know how they data is spread out inside the blue blob. 
+
+We saw previously that 
+histograms can help when we have saturated plots.
+But if we were to make a histogram
+of each column separately
+we would have two one-dimensional histogram 
+but no information of how these relate to eachother.
+
+Instead,
+what we need here is a single two-dimensional histogram
+of the scatter plot above.
+
+---
+
+## A heatmap allows us to visualize the relationship between two distributions
+
+```{python}
+alt.Chart(movies_extended).mark_rect().encode(
+    alt.X('Production Budget', bin=alt.Bin(maxbins=60)),
+    alt.Y('Worldwide Gross', bin=alt.Bin(maxbins=60)),
+    alt.Tooltip('Worldwide Gross', bin=alt.Bin(maxbins=60)),
```

Comment:
`Tooltip()` issue! Need to remove this or explain it! Thanks!

Reply:
Good catch, I think I was just playing around, removed